### PR TITLE
executor: init the final recv channel of hash agg with enough chunk

### LINKED ISSUE #13806: When sending data to HashAgg's father node, the execution logic is not paralleled
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

Find the Q17/Q18 is too much slower than others.

2. What did you expect to see?

Skip this chapter.

3. What did you see instead?

Testing TPC-H with scale factor 100.

For Q17, there're three tables joined, and one aggregation performed on the largest table `lineitem`. It executes 6mins+
For Q18, there're four tables joined, and one aggregation performed on the largest table `lineitem`. It executes 10mins+
For Q5, all tables are joined and the largest table `lineitem` was full joined without data filtered out before join, it executes 84.79s.
For Q9, nearly all tables are joined and one aggregation performed on join result, it executes 173.5s.

Though we cannot compre the execution time directly in fact. But it's very strange that the hash aggregation is too much slower compared with hash join.

So we looked into the codes and find something is unexpected.

```
	for !chk.IsFull() {
		e.finalInputCh <- chk
		result, ok := <-e.finalOutputCh
		if !ok { // all finalWorkers exited
			e.executed = true
			if chk.NumRows() > 0 { // but there are some data left
				return nil
			}
			if e.isChildReturnEmpty && e.defaultVal != nil {
				chk.Append(e.defaultVal, 0, 1)
			}
			e.isChildReturnEmpty = false
			return nil
		}
		if result.err != nil {
			return result.err
		}
		if chk.NumRows() > 0 {
			e.isChildReturnEmpty = false
		}
```
Here we do nothing about the `result.chk` received from the `e.finalOutputCh`, return the `chk` directly, which seems that there's only one chunk in the `e.finalInputCh`.

And i find that we do only send chunk to the `e.finalInputCh` here.

Though there're `FinalConcurrency`s FinalWorker waiting to send data to the `e.finalOutputCh`. But they're competing for only one chunk from `e.finalInputCh`, which make the parallel execution unparalled.

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

current master

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Fix #13806 

### What is changed and how it works?

Init the channel with enough chunk.

### Check List 

Tests 

 - Exsiting tests

### CODE REVIEW

File: executor/aggregate.go [hunk 1]
```diff
@@ -652,28 +655,28 @@ func (e *HashAggExec) parallelExec(ctx context.Context, chk *chunk.Chunk) error
...
 		return nil
 	}
-	for !chk.IsFull() {
-		e.finalInputCh <- chk
+	for {
 		result, ok := <-e.finalOutputCh
-		if !ok { // all finalWorkers exited
+		if !ok {
 			e.executed = true
-			if chk.NumRows() > 0 { // but there are some data left
```
Comment (winoros) [inline executor/aggregate.go]: If there's remaining data in `chk`. The `e.isChildReturnEmpty` must be false. Thus we don't need to return it here.

File: executor/aggregate.go [hunk 2]
```diff
@@ -668,28 +671,28 @@ func (e *HashAggExec) parallelExec(ctx context.Context, chk *chunk.Chunk) error
...
 			if e.isChildReturnEmpty && e.defaultVal != nil {
 				chk.Append(e.defaultVal, 0, 1)
 			}
-			e.isChildReturnEmpty = false
 			return nil
 		}
 		if result.err != nil {
 			return result.err
 		}
+		chk.Append(result.chk, 0, result.chk.NumRows())
```
Comment (SunRunAway) [inline executor/aggregate.go]: If `chk.NumRows == 0 && result.chk.NumRows() > chk.RequiredRows()`, we may use `chk.SwapColumn` to avoiding data copy.

Comment (winoros) [reply executor/aggregate.go]: We don't need to check `result.chk.NumRows() >= chk.RequiredRows()`? I think just check `chk.NumRow() == 0` is enough?

Comment (winoros) [reply executor/aggregate.go]: After some disscution with @XuHuaiyu and @yangxuanjia . We dicide to return the chunk directly. Ignore the `RequiredRows` from the upper executor.

**Comment by SunRunAway** [comment]
Should we cherry-pick this PR to 3.0?

**Comment by winoros** [comment]
I'm not sure. It's performance problem instead of bug. What's your opinion?

**Comment by qw4990** [comment]
> I'm not sure. It's performance problem instead of bug. What's your opinion?

emm, I also think it's better to pick it to v3.0.

**Pull request merged**